Validation/MuonGEMHits/src/GEMTrackMatch.cc

```diff
@@ -139,24 +139,28 @@ bool GEMTrackMatch::isSimTrackGood(const SimTrack &t)
 
 void GEMTrackMatch::buildLUT(const int maxChamberId)
 {
-
+  edm::LogInfo("GEMTrackMatch")<<"GEMTrackMatch::buildLUT"<<std::endl;
   edm::LogInfo("GEMTrackMatch")<<"max chamber "<<maxChamberId<<"\n";
 
-  std::vector<int> pos_ids;
-  pos_ids.push_back(GEMDetId(1,1,1,1,maxChamberId,useRoll_).rawId());
-
-  std::vector<int> neg_ids;
-  neg_ids.push_back(GEMDetId(-1,1,1,1,maxChamberId,useRoll_).rawId());
-
-  // VK: I would really suggest getting phis from GEMGeometry
-
+  std::vector<int> pos_ids, neg_ids;
   std::vector<float> phis;
-  phis.push_back(0.);
-  for(int i=1; i<maxChamberId+1; ++i)
-  {
-    pos_ids.push_back(GEMDetId(1,1,1,1,i,useRoll_).rawId());
-    neg_ids.push_back(GEMDetId(-1,1,1,1,i,useRoll_).rawId());
-    phis.push_back(i*10.);
+
+  for(auto it : gem_geom_->chambers()) {
+    if(it->id().region()>0) {
+      pos_ids.push_back(it->id().rawId());
+      edm::LogInfo("GEMTrackMatch")<<"added id = "<<it->id()<<" = "<<it->id().rawId()<<" to pos ids"<<std::endl;
+    }
+    if(it->id().region()<0) {
+      neg_ids.push_back(it->id().rawId());
+      edm::LogInfo("GEMTrackMatch")<<"added id = "<<it->id()<<" = "<<it->id().rawId()<<" to neg ids"<<std::endl;
+      const BoundPlane& bSurface(it->surface());
+      LocalPoint  lCentre( 0., 0., 0. );
+      GlobalPoint gCentre(bSurface.toGlobal(lCentre));
+      int cphi(static_cast<int>(gCentre.phi().degrees()));
+      if (cphi < 0) cphi += 360;
+      phis.push_back(cphi);
```

Comment:
How can you make the pair with pos_ids at L165 if  "phis" is only filled up for the negative part?